fdbserver/storageserver.actor.cpp

```diff
@@ -3318,6 +3336,11 @@ ACTOR Future<Void> getKeyValuesQ(StorageServer* data, GetKeyValuesRequest req)
 
 		Version commitVersion = getLatestCommitVersion(req.ssLatestCommitVersions, data->tag);
 		state Version version = wait(waitForVersion(data, commitVersion, req.version, span.context));
```

Comment:
You are right. My fix is wrong, and this needs more thought.

Here is what happened:
```
client requests read at 587602323
497.458019 CCD ID=f37f6cdeafd85582 Version=587602323 TS

SS converts read version to 581558882 in waitForVersion() for reads
497.460364 VVV ID=0ddff8edefbcfeef CommitVersion=565913092 Version=581558882 ReqVersion=587602323 VV=[{0:0,565913092} maxversion: 565913092] DebugID=f37f6cdeafd85582 BK,DD,MS,RK,RV,SS,TL

The readRange() throws transaction_too_old because storage version slightly increased:
497.462264 TTO1 ID=0ddff8edefbcfeef StorageVersion=582054715 Version=581558882 Range=\xff\x02/blog/\xae\xf4\xaa\xb8\xe7\
xf4\xee\x94\xa9G\xb0\x9e2\xbd\x85\x15\x17,\xe0\x7fq\xc4\xd2\xdb - \xff\x02/blog/\xae\xf4\xaa\xb8\xe7\xf4\xee\x94\xa9G\xb
0\x9e2\xbd\x85J BK,DD,MS,RK,RV,SS,TL
```

This repeats and client read never succeeded.

Reply:
>SS converts read version to 581558882 in waitForVersion() for reads
497.460364 VVV ID=0ddff8edefbcfeef CommitVersion=565913092 Version=581558882 ReqVersion=587602323 

Is 581558882 the oldest version at the time of waitForVersion() invocation?

Reply:
> Is 581558882 the oldest version at the time of waitForVersion() invocation?

Yes.